fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -619,11 +618,13 @@ def _validate_input(self, X, y, A, reinitialize=False):
             logger.warning("Setting sensitive_features to zeros")
             A = zeros(len(y))
 
+        if not self.skip_validation:
+            check_consistent_length(X, A)
+
         if (not is_fitted) or (reinitialize):
             self.__setup(X, y, A)
 
         if not self.skip_validation:
-            y = self._y_transform.transform(y)
             A = self._sf_transform.transform(A)
```

Comment:
what does validation have to do with transforming data? confused

Reply:
this is how it was when I found it, do you think it is unnecessary? I can't know why it was added in. I will have to change this when I do the changing of the categorical labels up too, so I can check it out more in detail.